stm32-modules/heater-shaker/simulator/comm_thread.cpp

```diff
@@ -39,16 +40,37 @@ auto comm_thread::build()
 }
 
 auto comm_thread::handle_input(tasks::Tasks<SimulatorMessageQueue>& tasks)
-    -> void {
-    auto linebuf = std::make_shared<std::string>(1024, 'c');
-    while (true) {
-        if (!std::cin.getline(linebuf->data(), linebuf->size() - 1, '\n')) {
-            return;
+-> void {
+    boost::asio::io_service io_context;
+
+    boost::asio::ip::tcp::socket mysocket(io_context);
+    boost::asio::ip::tcp::endpoint endpoint(
+            boost::asio::ip::address::from_string("127.0.0.1"), 9999);
+    boost::system::error_code ec;
+    mysocket.connect(endpoint, ec);
+    if (ec) {
+
```

Comment:
this can be 
```
std::cerr << "Failed to create socket: " << ec.category().name() << ": " << ec.value() << std::endl;
exit(ec.value()); // this should be from <cstdlib>
```